packages/studio/src/components/PageSettingsButton/EntityPageModal.tsx

```diff
@@ -64,9 +77,28 @@ export default function EntityPageModal({
         updateGetPathValue(pageName, getPathValue);
       }
       updateStreamScope(pageName, StreamScopeParser.parseStreamScope(form));
+      try {
+        const mapping = await generateTestData();
```

Comment:
Could we guard this behavior by first checking if the Stream Scope was edited? If the Scope wasn't modified (maybe it was just the URL), we shouldn't need to regenerate `localData` or `refreshActivePageEntities`, right?